src/js/core/methods/helpers/ethereumTypeDataConversions.js

```diff
@@ -0,0 +1,178 @@
+const paramTypeArray = new RegExp(/^(.*)\[([0-9]*)\]$/);
+const paramTypeBytes = new RegExp(/^bytes([0-9]*)$/);
+const paramTypeNumber = new RegExp(/^(u?int)([0-9]*)$/);
+
+/**
+ * Parse the given EIP-712 array type into its entries, and its length (if not dynamic)
+ * E.g. `uint16[32]` will return `{entryTypeName: 'uint16', arraySize: 32}`.
+ */
+export function parseArrayType(arrayTypeName: string): {
+    entryTypeName: string,
+    arraySize: number | null,
+} {
+    const arrayMatch = paramTypeArray.exec(arrayTypeName);
+    if (arrayMatch === null) {
+        throw ERRORS.TypedError(
+            'Method_InvalidParameter',
+            `typename ${arrayTypeName} could not be parsed as an EIP-712 array`,
+        );
+    }
+    const [_, entryTypeName, arraySize] = arrayMatch;
+    return {
+        entryTypeName,
+        arraySize: parseInt(arraySize, 10) || null,
+    };
+}
+
+/**
+ * Converts a number to a two's compliment representation.
+ *
+ * E.g. -128 would be 0x80 in two's compliment, while 127 would be 0x7F.
+ *
+ * BigNumber.js has no built-in function, unlike https://www.npmjs.com/package/bn.js
+ */
+function twosCompliment(number: BigNumber, bytes: number): BigNumber {
+    if (bytes < 1 || bytes > 32) {
+        throw new Error('Int byte size must be between is 1 and 32');
+    }
+    // Determine value range
+    const minValue = new BigNumber(2).exponentiatedBy(bytes * 8 - 1).negated();
+    const maxValue = minValue.negated().plus(1);
+
+    const bigNumber = new BigNumber(number);
+
+    if (bigNumber.isGreaterThan(maxValue) || bigNumber.isLessThan(minValue)) {
+        throw new Error(`Overflow when trying to convert number ${number} into ${bytes} bytes`);
+    }
+
+    if (bigNumber.isPositive()) {
+        return bigNumber;
+    }
+    return bigNumber.minus(minValue).plus(minValue.negated());
+}
+
+// $FlowIssue bigint-unsupported, TODO: Update flow when bigint is supported
+function intToHex(
+    number: number | bigint | BigNumber | string,
+    bytes: number,
+    signed: boolean,
+): string {
+    let bigNumber = new BigNumber(number);
+    if (signed) {
+        bigNumber = twosCompliment(bigNumber, bytes);
+    }
+    const hex = bigNumber.toString(16);
+    const hexChars = bytes * 2;
+    if (hex.length > hexChars) {
+        throw new Error(`Overflow when trying to convert number ${number} into ${bytes} bytes`);
+    }
+    return hex.padStart(bytes * 2, '0');
+}
+
+/**
+ * Encodes the given primitive data to a big-endian hex string.
+ *
+ * @param typeName - Primitive Solidity data type (e.g. `uint16`)
+ * @param data - The actual data to convert.
+ * @returns Hex string of the data.
+ */
+export function encodeData(typeName: string, data: any): string | Uint8Array {
+    if (paramTypeBytes.test(typeName) || typeName === 'address') {
+        return stripHexPrefix(data);
+    }
+    if (typeName === 'string') {
+        return new TextEncoder().encode(data);
```

Comment:
the codebase seems to use `Buffer.from('utf-8').toString('hex')`

Reply:
Done in 0103d17334c81e9ba7e1b7b3ed39c7b396f35ba7

Although, it does hurt slightly to use a Node.JS library instead of built-in tool in both Web/Node due to having to support Internet Explorer :sob: